pkg/segment/structs/segstructs.go

```diff
@@ -511,24 +519,89 @@ func (ss *SegStats) Init(rawSegStatJson []byte) error {
 	}
 	ss.IsNumeric = segStatJson.IsNumeric
 	ss.Count = segStatJson.Count
-	ss.Hll = hyperloglog.New()
-	err = ss.Hll.UnmarshalBinary(segStatJson.RawHll)
+	err = ss.CreateHllFromBytes(segStatJson.RawHll, true)
 	if err != nil {
-		log.Errorf("SegStats.Init: Failed to unmarshal hyperloglog error: %v data: %v", err, string(segStatJson.RawHll))
+		log.Errorf("SegStats.Init: Failed to create new segmentio Hll from raw bytes. error: %v data: %v", err, string(segStatJson.RawHll))
 		return err
 	}
 	ss.NumStats = segStatJson.NumStats
 	return nil
 }
 
+// Creates a new segmentio Hll with the defined SegHllSettings.
+// If failed to create, it will return a default Hll
+func CreateNewSegmentioHll() *hll.Hll {
+	segHll, err := hll.NewHll(SegHllSettings)
+	if err != nil {
+		// fallback to default Hll
```

Comment:
do we want to log an error here ? the err log line still should not flood, since it would get created per segment

Reply:
We would also be creating this per block during the query time. This function is only used on the query side for the cases of `timechart` and `stats` with `cardinality` function

Reply:
ok got it